backend/src/test/java/dev/projects/backend/controller/PersonControllerTest.java

```diff
@@ -0,0 +1,310 @@
+package dev.projects.backend.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import dev.projects.backend.collection.Person;
+import dev.projects.backend.enums.LoginRole;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class PersonControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Test
+    @DirtiesContext
+    @WithMockUser
+    void testCheckAndCreateMasterAdmin() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/person/master-admin")
+                        .with(csrf()))
+                .andExpect(status().is(201));
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/persons/role")
+                .param("role", LoginRole.ADMIN.toString()))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
+                        [
+                        {
+                                "userName": "${ENV_UN_Admin}",
+                                "passWord": "${ENV_PW_Admin}",
+                                "loginRole": "ADMIN"
+                              }
+                             ]
+                        """))
+                .andExpect(jsonPath("$[0].personId").isNotEmpty());
+    }
+
```

Comment:
missing test for "MasterAdmin already exists." ?

Reply:
position has been deleted